sdk/core/azure-core/src/main/java/com/azure/core/util/FluxUtil.java

```diff
@@ -98,8 +99,34 @@ private static void accept(ByteArrayOutputStream byteOutputStream, ByteBuffer by
      * @return The response from service call
      */
     public static <T> Mono<T> withContext(Function<Context, Mono<T>> serviceCall) {
+        return withContext(serviceCall, Collections.emptyMap());
+    }
+
+    /**
+     * This method converts the incoming {@code subscriberContext} from {@link reactor.util.context.Context Reactor
+     * Context} to {@link Context Azure Context}, adds the specified context attributes and calls the given lambda
+     * function with this context and returns a single entity of type {@code T}
+     * <p>
+     * If the reactor context is empty, {@link Context#NONE} will be used to call the lambda function
+     * </p>
+     *
+     * @param serviceCall serviceCall The lambda function that makes the service call into which azure context
+     * will be passed
+     * @param contextAttributes The map of attributes sent by the calling method to be set on {@link Context}.
+     * @param  <T> The type of response returned from the service call
+     * @return The response from service call
+     */
+    public static <T> Mono<T> withContext(Function<Context, Mono<T>> serviceCall,
+        Map<String, String> contextAttributes) {
```

Comment:
Just curious, under what scenarios, we would expect this map to contain more than one value ?

Reply:
Yes, it is mostly for future-proofing. So that in future if there are more tracing attribute requirements its easily extendable that way.